File: iOS 7 by tutorials/Chapter 19 - JavaScript Core/SpeechSynthesizerDemo/SpeechSynthesizerDemo/ViewController.m
//
//  ViewController.m
//  SpeechSynthesizerDemo
//
//  Created by lanjing on 15/7/20.
//  Copyright (c) 2015年 WaterWood. All rights reserved.
//

#import "ViewController.h"
#import "HLSpeechSynthesizerManager.h"

@interface ViewController ()

@end

@implementation ViewController
{
    HLSpeechSynthesizerManager *_speechSynthesizerManager;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    _speechSynthesizerManager = [[HLSpeechSynthesizerManager alloc] init];
    [_speechSynthesizerManager addVoiceSegmentWithString:@"this is a method"];
    [_speechSynthesizerManager addVoiceSegmentWithString:@"another method again"];

    
    [_speechSynthesizerManager addVoiceSegmentWithString:@"从明天起，from tomorrow, 做一个幸福的人"];
    [_speechSynthesizerManager addVoiceSegmentWithString:@"再见"];
    
    
    
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

-(void)dealloc
{
    
}

@end

File: iOS 8 by tutorials/Chapter 26 - iOS 8 Visual Effects/VisualEffectTest/VisualEffectTest/ViewController.swift
//
//  ViewController.swift
//  VisualEffectTest
//
//  Created by lanjing on 3/10/16.
//  Copyright © 2016 lanjing. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    @IBOutlet weak var imageView: UIImageView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


    @IBAction func blur(sender: UIBarButtonItem) {
        let blurEffect = UIBlurEffect(style: .Light)
  
        let blurEffectView = UIVisualEffectView(effect: blurEffect)
        blurEffectView.frame = imageView.frame
        
//        let vibrancyEffect = UIVibrancyEffect(forBlurEffect: blurEffect)
//        let vibrancyEffectView = UIVisualEffectView(effect: vibrancyEffect)
//        vibrancyEffectView.frame = imageView.frame
//        vibrancyEffectView.contentView.addSubview(view)
        
//        blurEffectView.contentView.addSubview(imageView)
//        view.addSubview(vibrancyEffectView)
//        view.addSubview(blurEffectView)
        view.insertSubview(blurEffectView, atIndex: 1)
//        blurEffectView.contentView.addSubview(imageView)
       
        
    }
}

